### PR TITLE
[BugFix] fix compatibility issue when collecting query statistics

### PR DESCRIPTION
**Pull request by silverbullet233**
this compatibility issue was introduce by https://github.com/StarRocks/starrocks/pull/27779

Before https://github.com/StarRocks/starrocks/pull/27779, pipeline engine didn't maintain scan statistics for each table, `QueryStatistics::_stats_items` was always empty, so _stats_items needn't be merged in `QueryStatistics::merge`.
But after that, `QueryStatistics::_stats_items` is no longer empty. In the implementation of older version, `QueryStatistics::_stats_items` is a vector and the problem of deduplication is not considered, which may occupy a lot of memory in `QueryStatistics::merge`. If a cluster is in the process of grayscale upgrade, the memory usage of the old version BE may increase.

A new session variable `enable_collect_table_level_scan_stats` is introduced here to control whether to collect table level scan stats, which is false by default, so that the behavior of query stats will remain the same as before during the grayscale upgrade. After all nodes are upgraded, we can set  `enable_collect_table_level_scan_stats` back to true to make fe scan metrics work normally.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 8 / 9 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver.cpp|6|7|85.71%|[263]|
|:large_blue_circle:|src/exec/pipeline/stream_pipeline_driver.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Review by stdpain — DISMISSED** [review]

**Review by satanson — DISMISSED** [review]

**Review by liuyehcf — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -1225,6 +1227,11 @@ public void setEnableParallelMerge(boolean enableParallelMerge) {
     @VarAttr(name = ENABLE_COUNT_STAR_OPTIMIZATION, flag = VariableMgr.INVISIBLE)
     private boolean enableCountStarOptimization = true;
 
+    // This variable is introduced to solve compatibility issues/
+    // see more details: https://github.com/StarRocks/starrocks/pull/29678
+    @VarAttr(name = ENABLE_COLLECT_TABLE_LEVEL_SCAN_STATS)
+    private boolean enableCollectTableLevelScanStats = false;
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:1233]: true?

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver.cpp|6|8|75.00%|[263, 805]|
|:large_blue_circle:|src/runtime/runtime_state.h|3|3|100.00%|[]|
|:large_blue_circle:|src/exec/pipeline/stream_pipeline_driver.cpp|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 1 above)
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java]: > true?

fixed

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver.cpp|6|8|75.00%|[263, 805]|
|:large_blue_circle:|src/runtime/runtime_state.h|3|3|100.00%|[]|
|:large_blue_circle:|src/exec/pipeline/stream_pipeline_driver.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver.cpp|6|8|75.00%|[263, 805]|
|:large_blue_circle:|src/runtime/runtime_state.h|3|3|100.00%|[]|
|:large_blue_circle:|src/exec/pipeline/stream_pipeline_driver.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver.cpp|6|8|75.00%|[263, 805]|
|:large_blue_circle:|src/runtime/runtime_state.h|3|3|100.00%|[]|
|:large_blue_circle:|src/exec/pipeline/stream_pipeline_driver.cpp|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver.cpp|6|8|75.00%|[263, 805]|
|:large_blue_circle:|src/runtime/runtime_state.h|3|3|100.00%|[]|
|:large_blue_circle:|src/exec/pipeline/stream_pipeline_driver.cpp|2|2|100.00%|[]|

**Review by sevev — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**